tests/tensor/rewriting/test_subtensor.py

```diff
@@ -2402,3 +2402,24 @@ def test_local_blockwise_advanced_inc_subtensor(set_instead_of_inc):
     else:
         expected_out[:, :, core_idxs] += test_y
     np.testing.assert_allclose(fn(test_x, test_y), expected_out)
+
+
+@pytest.mark.parametrize("fstop, lstop, lstep", [(None, 9, 1), (-1, -1, -1)])
+def test_slice_canonicalize(fstop, lstop, lstep):
+    x = tensor(shape=(3, 5, None, 9))
+    y = x[0:fstop, 0:5, 0:7, 0:lstop:lstep]
+    f = pytensor.function([x], y)
```

Comment:
If you compile a pytensor function you can evaluate it once with some numerical inputs to confirm it gives the expected numerical outputs. This is independent of checking the graph is the expected one.